Source/santactl/Commands/SNTCommandMetrics.m

```diff
@@ -0,0 +1,180 @@
+        values[index][@"created"] = [formatter stringFromDate:values[index][@"created"]];
+        values[index][@"last_updated"] = [formatter stringFromDate:values[index][@"last_updated"]];
+      }];
+    }
+  }
+
+  return mutableMetrics;
+}
+
+- (void)runWithArguments:(NSArray *)arguments {
+  __block NSDictionary *metrics;
+
+  dispatch_group_t group = dispatch_group_create();
+  dispatch_group_enter(group);
+
+  [[self.daemonConn remoteObjectProxy] metrics:^(NSDictionary *exportedMetrics) {
+    metrics = exportedMetrics;
+    dispatch_group_leave(group);
+  }];
+
+  // Wait a maximum of 5s for metrics collected from daemon to arrive.
+  if (dispatch_group_wait(group, dispatch_time(DISPATCH_TIME_NOW, NSEC_PER_SEC * 5))) {
+    fprintf(stderr, "Failed to retrieve some stats from daemon\n\n");
+  }
+
+  BOOL exportMetrics = [[SNTConfigurator configurator] exportMetrics];
+  NSURL *metricsURLStr = [[SNTConfigurator configurator] metricURL];
+  SNTMetricFormatType metricFormat = [[SNTConfigurator configurator] metricFormat];
+  NSUInteger metricExportInterval = [[SNTConfigurator configurator] metricExportInterval];
+  NSDictionary *normalizedMetrics = [self normalize:metrics];
+
+  if ([arguments containsObject:@"--json"]) {
+    // Format
+    NSData *metricData = [NSJSONSerialization dataWithJSONObject:normalizedMetrics
+                                                         options:NSJSONWritingPrettyPrinted
+                                                           error:nil];
+    NSString *metricStr = [[NSString alloc] initWithData:metricData encoding:NSUTF8StringEncoding];
+    printf("%s\n", [metricStr UTF8String]);
+  } else {
+    if (exportMetrics) {
+      printf(">>> Metrics Info\n");
+      printf("  %-25s | %s\n", "Metrics Server", [metricsURLStr.absoluteString UTF8String]);
+      printf("  %-25s | %s\n", "Metrics Format",
+             [[self stringFromMetricFormat:metricFormat] UTF8String]);
+      printf("  %-25s | %lu\n", "Export Interval (seconds)", metricExportInterval);
+    } else {
+      printf("Metrics not configured\n");
+      return;
+    }
+
+    printf(">>> Metrics Info\n\n");
+    printf(">>>\t Root Labels\n");
+
+    for (NSString *label in normalizedMetrics[@"root_labels"]) {
+      const char *labelStr = [label cStringUsingEncoding:NSUTF8StringEncoding];
+      const char *valueStr =
+        [normalizedMetrics[@"root_labels"][label] cStringUsingEncoding:NSUTF8StringEncoding];
+
+      printf("  %-25s | %s\n", labelStr, valueStr);
+    }
+
+    printf("\n");
+    printf(">>> Metrics \n");
+
+    for (NSString *metricName in normalizedMetrics[@"metrics"]) {
+      NSDictionary *metric = normalizedMetrics[@"metrics"][metricName];
+      const char *metricNameStr = [metricName UTF8String];
+      const char *description = [metric[@"description"] UTF8String];
+      // NSString * metricType =  makeSNTMetricTypeString(metric[@"type"]);
```

Comment:
Should this be uncommented?

Reply:
Yeah I had a minor tool failure with sourcetree should be fixed.